scalafmt-interfaces/native/src/main/java/org/scalafmt/interfaces/Scalafmt.scala

```diff
@@ -0,0 +1,143 @@
+package org.scalafmt.interfaces
```

Comment:
no reply

Reply:
I tried manually comparing the two versions with decompiled old java implementations and new scala ones, and the biggest problem I am seeing right now is the scala varargs (*) not  being compatible with java ones (...), the scala one gets compiled into a method with scala.Seq, which probably cannot be used from java as varargs. Also every class in scala gets annotated with a `scala.reflect.ScalaSignature` annotation, but I am not sure how much trouble that would cause (I imagine that would require the build tool plugins to also have that class on the classpath, breaking compatibility by now requiring a new scala dependency, but I am unsure how class loading would interact with it). I would say it's definitely safer to have the 2 implementations, if only a little bit more annoying probably